sources/core/Xenko.Core/Threading/ThreadPool.cs

```diff
@@ -19,109 +20,71 @@ internal class ThreadPool
         public static readonly ThreadPool Instance = new ThreadPool();
 
         private readonly int maxThreadCount = Environment.ProcessorCount + 2;
-        //private readonly int maxThreadCount = Environment.ProcessorCount * 2;
-        private readonly List<Task> workers = new List<Task>();
         private readonly Queue<Action> workItems = new Queue<Action>();
         private readonly ManualResetEvent workAvailable = new ManualResetEvent(false);
 
-        private SpinLock spinLock = new SpinLock();
-        private int activeThreadCount;
+        private object _lock = new object();
+        volatile private int activeThreadCount;
+        volatile private int aliveThreadCount;
 
         public void QueueWorkItem([NotNull] [Pooled] Action workItem)
         {
-            var lockTaken = false;
-            try
+            lock (_lock)
             {
-                spinLock.Enter(ref lockTaken);
-
                 PooledDelegateHelper.AddReference(workItem);
                 workItems.Enqueue(workItem);
-
-                if (activeThreadCount + 1 >= workers.Count && workers.Count < maxThreadCount)
+                workAvailable.Set();
+                if (activeThreadCount + 1 >= aliveThreadCount && aliveThreadCount < maxThreadCount)
```

Comment:
Usually, it is an issue to access a field with and without `Interlocked`.

Reply:
Wait, I thought that reading a 32bit value was guaranteed to be atomic ? I just made them volatile to guide the compiler ?